jsonrpc/eth_endpoint.go

```diff
@@ -330,12 +345,21 @@ func (e *Eth) EstimateGas(
 	// If the sender address is present, recalculate the ceiling to his balance
 	if transaction.From != types.ZeroAddress && transaction.GasPrice != nil && gasPriceInt.BitLen() != 0 {
 		// Get the account balance
+
+		// If the account is not initialized yet in state,
+		// assume it's an empty account
+		accountBalance := big.NewInt(0)
 		acc, err := e.d.store.GetAccount(header.StateRoot, transaction.From)
-		if err != nil {
+		if err != nil && !errors.Is(err, ErrStateNotFound) {
```

Comment:
```suggestion
		if !errors.Is(err, ErrStateNotFound) {
```

Reply:
No need to omit the `err != nil` check here, as we need to check that in fact the error is `nil` later:

https://github.com/0xPolygon/polygon-sdk/blob/33e84c77e253db62ef35f12f49e0fa9c385db252/jsonrpc/eth_endpoint.go#L356-L360